grpc.go

```diff
@@ -359,6 +359,27 @@ func (a *agentGRPC) CreateContainer(ctx context.Context, req *pb.CreateContainer
 		return emptyResp, err
 	}
 
+	ctr := &container{
+		id:        req.ContainerId,
+		processes: make(map[string]*process),
+		mounts:    mountList,
+	}
+
+	a.sandbox.setContainer(req.ContainerId, ctr)
+
+	// Make sure we rollback the mounts performed in case the container
+	// creation failed.
+	defer func() {
+		if err != nil {
+			if ctr.container != nil {
+				ctr.container.Destroy()
+			}
+
+			a.sandbox.deleteContainer(req.ContainerId)
+			removeMounts(ctr.mounts)
```

Comment:
Shouldn't that be:

```
err = removeMounts(ctr.mounts)
```

Reply:
Well not really since I don't think an error from the rollback part executed from `defer` should not gate anything.

Reply:
We're already in the error path but we should probably atleast combine the errors so they get logged at a higher level since we don't expect.

Reply:
This looks overkill to me... We're already into the error path, meaning an error is going to be logged, but you also want the potential error of the error ? What should I log ? A `Debug()` one ?

Reply:
I can see that point of view and I also see that it's going to complicate the code a little, but why are we doing rollback? Answer: to ensure we *undo* the operations the agent just performed.. So we shouldn't fully ignore operations that return errors imho as that would hide bugs relating to the undo.

As a minimum, I think we should do something like:

```
err2 := removeMounts(ctr.mounts)
if err2 != nil {
    agentLog.WithError(err2).Error("rollback failed")
}
```

That way atleast we get something useful in the logs.

btw, the naming is a little random - it's not very obvious at first glance that the `defer` basically undoes `addStorages()`. Maybe the code block could be put into a `removeStorages()` to make that clearer?

Reply:
Yeah makes sense, let me rework a little bit